### ISSUE TITLE
NaN loss when training ResNet3D

### ISSUE THREAD

**Issue by MounirB** [JihongJu/keras-resnet3d#9]
Hello,

I am encountering a problem of NaN loss when training your architecture on a dataset of videos (is it a symptom of Exploding/Vanishing gradients ?). 
I used **Adam(lr=1e-3)** as an optimisation function.

To tackle this, I'm trying different approaches :  
- Setting the **kernel_initializer** to default initialization (**glorot_uniform** distribution).
- Removing the l2 regularization and the reg_factor (even if I incur an overfitting problem).
My next move if it doesn't tackle the problem : 
- Changing the categorical cross entropy loss by a mean squared error.

What do you think about my problem ? Do you have any other suggestions ?

**Comment by yingji425** [comment]
Hello, have you solved this problem? i am facing the same situation, my loss is always NaN, and I don't know how to fix it. Thank you

**Comment by MounirB** [comment]
I followed up by asking this question on [stackexchange](https://datascience.stackexchange.com/questions/85310/the-association-of-the-l2-regularization-and-the-he-normal-weights-initializatio). However, I gave up since I didn't find any solution.
